### ISSUE TITLE
Cannot read property 'toLowerCase' of null

### ISSUE THREAD

**Issue by lldata** [Huachao/vscode-restclient#219]
- VSCode Version:  1.22.2
- OS Version: win7
- REST Client Version: 0.18.4

![image](https://user-images.githubusercontent.com/5525242/41086231-ac2668f0-6a39-11e8-84ca-58c61a426d68.png)

After running VS Code REST client for a while, I get this error when trying to call any REST service.
"Cannot read property 'toLowerCase' of null".

I can't reproduce it. It just happens every couple of hours or days.
Closing the http file does not resolve it. Only restarting VSCode resolves the issue.

It is quite annoying. I'd like to provide more information, but I don't know where to start.
I get the feeling that the toLowerCase hides another error.

![image](https://user-images.githubusercontent.com/5525242/41086604-c92f16c6-6a3a-11e8-8b9e-033eab64f15d.png)

**Comment by Huachao** [comment]
@lldata will this only occur when you are sending a specific request?

**Comment by Huachao** [comment]
@lldata since retarting can work, maybe it is related to something that keep in memory of the extension lifetime. One thing I can think of is request variable cache. I will continue inspecting this tomorrow.

**Comment by lldata** [comment]
Sounds likely. I use request variables extensively. Something related to caching or other mutable state seems very likely. But it might be hard to fix, if you can't reproduce it ;-)

**Comment by Huachao** [comment]
@lldata I checked all the code used the `toLowerCase()` method, I can't find any risk code to call this function. And I want to confirm with you for some possibilities.

- Can you show me all the named requests'(requests that are referenced by other requests) response content type header value.
- What's your setting value for `rest-client.previewOption`?
- Do you use proxy setting `rest-client.excludeHostsForProxy`?

Thanks in advance.

**Comment by lldata** [comment]
- I have never noticed a response without Content-Type header. I am pretty sure we have that always.
- "rest-client.previewOption": "full" (default)
- Yes! We use a proxy server
- The errors only occurs when I click the request links

When I first got the error some time ago, I dived into the VSCode debugger and caught the error there.
To me it looks like vs code is trying to do a toLowerCase on something - presumably an error from a lower layer. I'll try to catch it again, when I see it next time.

**Comment by Huachao** [comment]
@lldata so do you mean the error actually happens in the code from VSCode, not from my extension directly?

**Comment by lldata** [comment]
From the debug session my impression was that the toLowerCase call was somewhere in the VSCode code base. But to me it looked like there was a different error below. 
Also it only happens when I use the rest-client and only when I click "Send request". 
I think it is safe to assume that the error is in your plugin. But it might not be in a toLowerCase call.

**Comment by Huachao** [comment]
@lldata how did you go into the debug session of vscode

**Comment by lldata** [comment]
I managed to reproduce it and get this Call Stack:

`e.onDidNotificationChange (notificationsAlerts.ts:39)
(anonymous) (notificationsAlerts.ts:28)
e.fire (event.ts:140)
e.notify (notifications.ts:127)
e.notify (notificationService.ts:65)
(anonymous) (mainThreadMessageService.ts:86)
e._showMessage (mainThreadMessageService.ts:47)
e.$showMessage (mainThreadMessageService.ts:41)
e._doInvokeHandler (rpcProtocol.ts:174)
e._invokeHandler (rpcProtocol.ts:159)
e._receiveRequest (rpcProtocol.ts:108)
e._receiveOneMessage (rpcProtocol.ts:90)
(anonymous) (rpcProtocol.ts:33)
(anonymous) (rpcProtocol.ts:213)
e.fire (event.ts:140)
(anonymous) (ipc.net.ts:81)
emitOne (events.js:96)
emit (events.js:191)
readableAddChunk (_stream_readable.js:178)
Readable.push (_stream_readable.js:136)
onread (net.js:560)`

![image](https://user-images.githubusercontent.com/5525242/41218826-b8f55d0e-6d5c-11e8-8335-cfb4ba95087c.png)

**Comment by lldata** [comment]
I just spent some time browsing around the debugger. It seems that the question is: how calls onread.
I think some kind of event/exception is emitted from some code that does not show up in the Call Stack.
"onread" in net.js is an event handler.

About the only useful thing I could find is that the event is from the restclient plugin:
![image](https://user-images.githubusercontent.com/5525242/41219801-c366fa1a-6d5f-11e8-9b5a-fc1eb1366aac.png)

**Comment by Huachao** [comment]
@lldata Uhh... Thanks for your help. And can you have a clear repro step for me? Like sending a request looks like what, and after that, everything no longer works until you restart vscode.

**Comment by lldata** [comment]
I am sorry. I can't reproduce it at will. 
It just happens - then I restart and it works again ..

**Comment by Huachao** [comment]
@lldata can you try to repro this issue in latest version 0.19.0, and I add some error log in output channel.

**Comment by lldata** [comment]
Thank you. I have updated and will investigate next time I can reproduce it.

**Comment by Huachao** [comment]
@lldata did you meet this problem recently?

**Comment by Huachao** [comment]
@lldata next time you meet the issue, please help to check the REST output channel of VSCode, more error details are logged there

**Comment by lldata** [comment]
We managed to reproduce it. My collegues report that they can reproduce it consistently by copying requests from Skype and into a .rest file.

Here is the output from the debug channel:
```
TypeError: Cannot read property 'toLowerCase' of null
TypeError: Cannot read property 'toLowerCase' of null
	at Function.ignoreProxy (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:238:45)
	at HttpClient.<anonymous> (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:158:53)
	at Generator.next (<anonymous>)
	at C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:7:71
	at new Promise (<anonymous>)
	at __awaiter (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:3:12)
	at HttpClient.prepareOptions (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:108:16)
	at HttpClient.<anonymous> (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:44:40)
	at Generator.next (<anonymous>)
	at C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:7:71
	at new Promise (<anonymous>)
	at __awaiter (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:3:12)
	at HttpClient.send (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\utils\httpClient.js:43:16)
	at RequestController.<anonymous> (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\controllers\requestController.js:126:55)
	at Generator.next (<anonymous>)
	at C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\controllers\requestController.js:13:71
	at new Promise (<anonymous>)
	at __awaiter (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\controllers\requestController.js:9:12)
	at RequestController.runCore (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\controllers\requestController.js:119:16)
	at RequestController.<anonymous> (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\controllers\requestController.js:94:24)
	at Generator.next (<anonymous>)
	at fulfilled (C:\Users\wql\.vscode\extensions\humao.rest-client-0.20.1\out\src\controllers\requestController.js:10:58)
	at <anonymous>

```

**Comment by Huachao** [comment]
@lldata Thanks so much @lldata. I can reproduce this issue as illustrated in the callstack, which needs to meet **three** preconditions:
1. Set the `http.proxy` setting
2. Set the `rest-client.excludeHostsForProxy` setting
3. The request url is not in a well defined format, like missing protocol, etc.,

And I will fix this issue ASAP

**Comment by Huachao** [comment]
@lldata I will publish this fix in next release.

**Comment by lldata** [comment]
Amazing feedback time :100: :+1: 

Thank you for your work!

**Comment by Huachao** [comment]
@lldata I have published the new version 0.20.3, and you can verify it. And if you still meet this issue, don't hesitate to contact me.
